mlflow/xgboost/__init__.py

```diff
@@ -718,27 +731,73 @@ def infer_model_signature(input_example):
                 model_format=model_format,
             )
 
+        # Whether to automatically log the training dataset as a dataset artifact.
+        if _log_datasets:
+            try:
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
+
+                # create a dataset
+                import xgboost as xgb
+
+                # dmatrix has a get_data method added in 1.7. skip for earlier versions.
+                if Version(xgb.__version__) >= Version("1.7.0"):
+                    data = dtrain.get_data()
```

Comment:
If the user specifies the `evals` parameter, which is a tuple of `(DMatrix, string)`, can we log information about these datasets as well? In this case, the string element in the tuple is the context.